src/main/java/org/eln2/mc/common/network/serverToClient/SingleDoubleElementGuiOpenPacket.kt

```diff
@@ -0,0 +1,49 @@
+package org.eln2.mc.common.network.serverToClient
+
+import net.minecraft.client.Minecraft
+import net.minecraft.core.BlockPos
+import net.minecraft.network.FriendlyByteBuf
+import net.minecraftforge.api.distmarker.Dist
+import net.minecraftforge.fml.loading.FMLEnvironment
+import net.minecraftforge.network.NetworkEvent
+import org.eln2.mc.common.cell.ISingleElementGuiCell
+import org.eln2.mc.common.containers.SingleValueCellContainer
+import java.util.function.Supplier
+
+class SingleDoubleElementGuiOpenPacket(val value: Double, val pos: BlockPos) {
+
+    companion object {
+        fun encode(packet: SingleDoubleElementGuiOpenPacket, buffer: FriendlyByteBuf) {
+            buffer.writeDouble(packet.value)
+            buffer.writeBlockPos(packet.pos)
+        }
+
+        fun decode(buffer: FriendlyByteBuf): SingleDoubleElementGuiOpenPacket {
+            return SingleDoubleElementGuiOpenPacket(buffer.readDouble(), buffer.readBlockPos())
+        }
+
+        fun handle(packet: SingleDoubleElementGuiOpenPacket, supplier: Supplier<NetworkEvent.Context>) {
+            val ctx = supplier.get()
+
+            ctx.enqueueWork {
+                if (Dist.CLIENT == FMLEnvironment.dist) {
+                    handleClient(packet, Minecraft.getInstance())
+                }
+            }
+        }
+
+        @Suppress("UNCHECKED_CAST")
```

Comment:
Generally speaking I'm a fan of putting these directly adjacent to the statement if possible.